### PR TITLE
logging: expose envoy_logging_callbacks

### PR DESCRIPTION
**Pull request by junr03**
Description: exposes new functionality in the bridge layer that will leverage #1354.
Risk Level: low
Testing: updated tests

Signed-off-by: Jose Nino <jnino@lyft.com>

### CODE REVIEW

File: library/common/engine.cc [hunk 1]
```diff
@@ -4,13 +4,15 @@
...
 #include "library/common/stats/utility.h"
 
 namespace Envoy {
 
-Engine::Engine(envoy_engine_callbacks callbacks, const char* config, const char* log_level,
+Engine::Engine(envoy_engine_callbacks callbacks, envoy_logging_callbacks logging_callbacks,
+               const char* config, const char* log_level,
                std::atomic<envoy_network_t>& preferred_network)
-    : callbacks_(callbacks) {
+    : callbacks_(callbacks), logging_callbacks_(logging_callbacks) {
```
Comment (goaway) [inline library/common/engine.cc:15]: logger_, imo

**Comment by goaway** [comment]
A few high level comments:

I’d advocate for renaming types representing “logger_callbacks” to just be “logger”, and “on_log” calls to just be “log”.

I don't really see the purpose for exposing a flush callback. Since it's ultimately up to the logger implementation to decide what it does with the logs, I don't think it really cares when Envoy decides to call flush in its internal implementations.

Not I'm a little concerned about the increase in things appearing in the Engine constructor. Another option would be to have this be runtime configurable. We don't need it right now, but we should maybe also think about the case where people also might want to add more than one logger.

**Comment by junr03** [comment]
> I don't really see the purpose for exposing a flush callback. Since it's ultimately up to the logger implementation to decide what it does with the logs, I don't think it really cares when Envoy decides to call flush in its internal implementations.

Discussed this. In addition, given that afaict Envoy only flushes on exit, we have decided to not expose that part of the API for now.

**Review by goaway — APPROVED** [review]
This looks good, thanks for making the changes!

**Pull request merged**